src/vm/ast/sugar_expander/mod.rs

```diff
@@ -0,0 +1,193 @@
+use std::convert::TryInto;
+use vm::representations::{PreSymbolicExpression, PreSymbolicExpressionType, SymbolicExpression, SymbolicExpressionType};
+use vm::types::{QualifiedContractIdentifier, Value, PrincipalData, StandardPrincipalData};
+use vm::ast::types::{ContractAST, BuildASTPass};
+use vm::ast::errors::{ParseResult, ParseError, ParseErrors};
+use vm::functions::NativeFunctions;
+
+pub struct SugarExpander {
+    issuer: StandardPrincipalData   
+}
+
+impl BuildASTPass for SugarExpander {
+
+    fn run_pass(contract_ast: &mut ContractAST) -> ParseResult<()> {
+        let pass = SugarExpander::new(contract_ast.contract_identifier.issuer.clone());
+        let expressions = pass.run(&mut contract_ast.pre_expressions);
+        contract_ast.expressions = expressions;
+        Ok(())
+    }
+}
+
+impl SugarExpander {
+
+    fn new(issuer: StandardPrincipalData) -> Self {
+        Self { issuer }
+    }
+
+    pub fn run(&self, pre_expressions: &[PreSymbolicExpression]) -> Vec<SymbolicExpression> {
+        self.transform(pre_expressions)
+    }
+
+    pub fn transform(&self, pre_expressions: &[PreSymbolicExpression]) -> Vec<SymbolicExpression> {
+        let mut expressions = Vec::new();
+        for pre_expr in pre_expressions.iter() {
+
+            let mut expr = match pre_expr.pre_expr {
+                PreSymbolicExpressionType::AtomValue(ref content) => {
+                    SymbolicExpression::atom_value(content.clone())
```

Comment:
I think this should be a `literal_value`. That will make it so that _all_ contract literals will be represented with LiteralValue, and then AtomValues would only be constructed by the interpreter at runtime (via evaluation of high-ordered functions like map, fold, or transaction invocation).

Reply:
Sounds good to me. I thought that we wanted to keep track of the contract-identifiers being unsugared, vs the others.

Reply:
Addressed with https://github.com/blockstack/blockstack-core/pull/1105/commits/c871057b86ad7ab2c64a84ba2986f42b469860ec